services/prometheus/prometheus.go

```diff
@@ -159,27 +159,41 @@ func (svc *Service) marshalConfig() ([]byte, error) {
 		if err != nil {
 			return errors.WithStack(err)
 		}
+
+		var rdsParams []*scrapeConfigParams
 		for _, str := range agents {
 			agent := str.(*models.Agent)
 			if agent.Disabled {
```

Comment:
looks like we can remove it by using where clause in sql script here https://github.com/percona/pmm-managed/pull/302/files#diff-f4853114e52d6ca4ddfc045924523970R158

Reply:
Added condition for disabled agents, and also for NULL listen_port.